samtranslator/swagger/swagger.py

```diff
@@ -799,26 +802,193 @@ def add_resource_policy(self, resource_policy):
         if resource_policy is None:
             return
 
+        iam_allow_list = resource_policy.get('IamAllowList')
+        iam_deny_list = resource_policy.get('IamDenyList')
+        ip_allow_list = resource_policy.get('IpAllowList')
+        ip_deny_list = resource_policy.get('IpDenyList')
+        source_vpc_allow_list = resource_policy.get('SourceVpcAllowList')
+        source_vpc_deny_list = resource_policy.get('SourceVpcDenyList')
         custom_statements = resource_policy.get('CustomStatements')
 
-        if custom_statements is not None:
-            if not isinstance(custom_statements, list):
-                custom_statements = [custom_statements]
+        if iam_allow_list is not None:
+            resource_list = self._get_method_path_uri_list(path, api_id, stage)
+            self._add_iam_resource_policy_for_method(iam_allow_list, "Allow", resource_list)
 
-            self.resource_policy['Version'] = '2012-10-17'
-            if self.resource_policy.get('Statement') is None:
-                self.resource_policy['Statement'] = custom_statements
-            else:
-                statement = self.resource_policy['Statement']
-                if isinstance(statement, list):
-                    statement.extend(custom_statements)
-                else:
-                    statement = [statement]
-                    statement.extend(custom_statements)
+        if iam_deny_list is not None:
```

Comment:
For this section (allow and deny lists): does order matter? Do we need to allow then deny or deny then allow? I've seen some policies that deny everything then only allow a few things, and I'm not quite sure how that works.

Reply:
I don't believe order matters for policy documents.

Reply:
Yeah looked into it- it appears order does not matter.